### ISSUE TITLE
Internal Server Error when using the Java EE Security API with Basic authentication without password

### ISSUE THREAD

**Issue by tofflos** [payara/Payara#2179]
# Description #
----------

I've tried using the new Java EE Security API included in Payara Micro 4.1.2.174. I have an BasicAuthenticationMechanismDefinition annotation and a class that implements IdentityStore. Authentication works as expected when providing a username and password. Payara responds with an Internal Server Error before reaching the validate method when a password is not provided.

## Expected Outcome

Authentication should reach the validate method even when the password is missing or at the very least report an Exception in the server log.

## Current Outcome

The server responds with Internal Server Error before reaching the validate method when a password is not provided.

## Steps to reproduce (Only for bug reports) 

1. Create a WAR project.
2. Add a class that extends Application
3. Add a JAXRS resource
4. Add a BasicAuthenticationMechanismDefinition annotation
5. Add a class that implements IdentityStore
6. Build and deploy the application
java -jar payara-micro-4.1.2.174.jar --deploy mavenproject01-1.0-SNAPSHOT.war
7. Connect to the application using username and password
curl http://localhost:8080//mavenproject01-1.0-SNAPSHOT/api/v1/items -H "Authorization: Basic NjY0NmYwYWEtYjJhOC00ODNhLTk0YWYtMDYzYWY1Nzk5MzE4OnRlc3Q="
Note response 200 OK.
8. Connect to the application using only username
curl http://localhost:8080//mavenproject01-1.0-SNAPSHOT/api/v1/items -H "Authorization: Basic NjY0NmYwYWEtYjJhOC00ODNhLTk0YWYtMDYzYWY1Nzk5MzE4Og=="
Note response 500 Internal Server Error.
Note missing log entry in server log.

## Environment ##

- **Payara Version**: 4.1.2.174
- **Edition**: Micro
- **JDK Version**: 8 u151 - OpenJDK
- **Operating System**: Windows
- **Database**: Not applicable

**Comment by tofflos** [comment]
Minimal project that reproduces the error:
[mavenproject01.zip](https://github.com/payara/Payara/files/1500271/mavenproject01.zip)

**Comment by arjantijms** [comment]
The offending fragment that causes the exception to be effectively swallowed is in

`RealmAdapter#validate`:

```java
             isValidateSuccess = SUCCESS.equals(authContext.validateRequest(messageInfo, subject, null));

                if (isValidateSuccess) { // store it only if validateRequest = true
                    storeInRequest(servletRequest, messageInfo, authContext);
                }
            } else {
                throw new AuthException("null ServerAuthContext");
            }
        } catch (AuthException ae) {
            _logger.log(FINE, "JMAC: http msg authentication fail", ae);
            servletResponse.setStatus(SC_INTERNAL_SERVER_ERROR);
        } catch (RuntimeException e) {
            _logger.log(FINE, "JMAC: Exception during validateRequest", e);
            servletResponse.sendError(SC_INTERNAL_SERVER_ERROR);
        }
```

Exceptions are logged, but at the FINE level.

Omitting the password should not cause a 500 of course. I'll check for that separately in Soteria. Thanks for the report!

**Comment by tofflos** [comment]
I'm not very familiar with the logging mechanism used in Payara and spent a few minutes figuring out how to set the log levels. Adding it here for future reference.

1. Run Payara with the rootdir argument. Payara will then create a new folder and extract its' internal configuration files including the one used for logging.

`java -jar payara-micro-4.1.2.174.jar --rootdir rootdir`

2. Figure out the name of the logger by looking at [LogDomains](
https://github.com/payara/Payara/blob/master/nucleus/common/common-util/src/main/java/com/sun/logging/LogDomains.java) and [RealmAdapter](https://github.com/payara/Payara/blob/master/appserver/security/webintegration/src/main/java/com/sun/web/security/RealmAdapter.java).

3. Append the name of the logger to the logging configuration file.

`echo "javax.enterprise.system.container.web.level=FINE" >> ./rootdir/config/logging.properties`

4. Launch Payara with the customized logging configuration file.

`java -jar payara-micro-4.1.2.174.jar --deploy mavenproject01-1.0-SNAPSHOT.war --logproperties  ./rootdir/config/logging.properties`

**Comment by smillidge** [comment]
Internal bug ID PAYARA-2412

**Comment by smillidge** [comment]
Fixed by #2407
